lib/volatile_hash.rb

```diff
@@ -12,6 +12,41 @@ def initialize(options)
         @refresh = options[:refresh] or false
     end
 
+    def key?(key)
+        value = @cache.key?(key)
+        if @strategy == 'ttl'
+            if @registry[key] && expired?(key)
+                @cache.delete key
+                @registry.delete key
+                value = false
+            end
+            if @refresh and @registry[key]
+                set_ttl(key)
+            end
+        else
+            lru_update key if @cache.has_key?(key)
+        end
+        value #in case of LRU, just return the value that was read
+    end
+
+    def keys
```

Comment:
Isn't this the same as Ruby's built-in .keys?

hash = {"apple" => "fruit", "carrot" => "vegetable"}
array = hash.keys   #=> ["apple", "carrot"]

Reply:
Yes, but with the added feature that it does not return keys that have expired